Test/Tests.cs

```diff
@@ -0,0 +1,260 @@
+            
+            string documentId = (string)postDocResponse["documentId"];
+            var response = client.PostPredictions(documentId,Example.ModelType());
+            
+            var expected = new List<string>(){"documentId", "predictions"} ;
+            JObject jsonResponse = JObject.Parse(response.ToString());
+            foreach (var field in jsonResponse)
+            {
+                Assert.IsTrue(expected.Contains(field.Key));
+                //Console.WriteLine($"Key: {field.Key}, Val: {field.Value.ToString()}" );
+            }
+        }
+
+        [Test]
+        public void TestPostDocumentId()
+        {
+            Console.WriteLine("TestPostDocumentId");
+            Client client = new Client(Example.Endpoint());
+            var postDocResponse = JsonSerialPublisher.ObjectToDict<Dictionary<string, string>>(client.PostDocuments(Example.ContentType(), Example.ConsentId()));
+            
+            client.PutDocument(Example.DocPath(),Example.ContentType(),postDocResponse["uploadUrl"]);
+            
+            var feedback = new List<Dictionary<string, string>>()
+            {
+                new Dictionary<string, string>(){{"label", "total_amount"},{"value", "54.50"}},
+                new Dictionary<string, string>(){{"label", "purchase_date"},{"value", "2007-07-30"}}
+            };
+            
+            var response = client.PostDocumentId(postDocResponse["documentId"], feedback);
+            var expected = new List<string>(){"documentId", "consentId", "uploadUrl", "contentType", "feedback"} ;
+            JObject jsonResponse = JObject.Parse(response.ToString());
+            foreach (var field in jsonResponse)
+            {
+                Assert.IsTrue(expected.Contains(field.Key));
+                //Console.WriteLine($"Key: {field.Key}, Val: {field.Value.ToString()}" );
+            }
+        }
+        
+        [Test]
+        public void TestDeleteConsentId()
+        {
+            Console.WriteLine("TestDeleteConsentId");
+            Client client = new Client(Example.Endpoint());
+            var postDocResponse = JsonSerialPublisher.ObjectToDict<Dictionary<string, string>>(client.PostDocuments(Example.ContentType(), Example.ConsentId()));
+            
+            client.PutDocument(Example.DocPath(),Example.ContentType(),postDocResponse["uploadUrl"]);
+            
+            var expected = new List<string>(){ "consentId", "documentIds"} ;
+            var response = client.DeleteConsentId(postDocResponse["consentId"]);
+            JObject jsonResponse = JObject.Parse(response.ToString());
+            
+            foreach (var field in jsonResponse)
+            {
+                Assert.IsTrue(expected.Contains(field.Key));
+                //Console.WriteLine($"Key: {field.Key}, Val: {field.Value.ToString()}" );
+            }
+        }
+
+        [Test]
+        public void TestHashSigning()
```

Comment:
I'm not quite sure which part of the signing process we are testing here. If we are testing AWS sig v4 Signature calculation, would a better test name incorporate some of these elements?

Reply:
This is just the low level hash calculation. Just to make sure that it is equal the python calculation.